bson/src/test/unit/org/bson/BsonHelper.java

```diff
@@ -85,11 +90,16 @@
     }
 
     public static List<BsonValue> valuesOfEveryType() {
-        return BSON_VALUES;
+        return getBsonValues();
     }
 
     public static BsonDocument documentWithValuesOfEveryType() {
-        return BSON_DOCUMENT;
+        BsonDocument document = new BsonDocument();
```

Comment:
And here's the root cause of the bad test: this method returns a singleton BsonDocument, which the test then goes ahead and mutates.  It wasn't caught because it happened to be the last test that used this method.